### PR TITLE
Record block data size in meta

### PR DESCRIPTION
**Pull request by mdisibio**
**What this PR does**:
This PR adds a new field to the block meta for the final data file size, and support for this and other recent fields to the tempo-cli. This PR is a precursor to switching compaction over to a target file size instead of target trace count.   

**Which issue(s) this PR fixes**:
n/a

**Checklist**
- [x] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: tempodb/backend/block_meta.go [hunk 1]
```diff
@@ -22,6 +22,8 @@ type BlockMeta struct {
 	StartTime       time.Time `json:"startTime"`
 	EndTime         time.Time `json:"endTime"`
 	TotalObjects    int       `json:"totalObjects"`
+	TotalSize       uint64    `json:"totalSize"`
+	FinalSize       uint64    `json:"finalSize"`
```
Comment (annanay25) [inline tempodb/backend/block_meta.go:26]: Why do we need two sizes in the block meta? Let's just use the compressed size?

File: tempodb/encoding/compactor_block.go [hunk 2]
```diff
@@ -121,6 +123,7 @@ func (c *CompactorBlock) BlockMeta() *backend.BlockMeta {
 
 	meta.StartTime = c.inMetas[0].StartTime
 	meta.EndTime = c.inMetas[0].EndTime
+	meta.FinalSize = c.totalBytesFlushed
```
Comment (joe-elliott) [inline tempodb/encoding/compactor_block.go:126]: Why not use `.DataLength()` in the compactor block like the complete block?

**Review by joe-elliott — COMMENTED** [review]
An appender is used by both the compactor and complete blocks.  It already does some record keeping w/r to the amount of data that has been appended.  If we want to track "totalSize" does it make sense to just put the logic there?

Echo @annanay25 's question about whether or not we need both.  the compressed size seems like enough to me.  If we are just interested in compression ratio we may want to consider metricing it.  I know Loki does that.

File: tempodb/backend/block_meta.go (hunk 1 above)
Comment (mdisibio) [reply tempodb/backend/block_meta.go]: Sounds like the consensus is that storing the uncompressed data size isn't valuable.  Removed.

File: tempodb/encoding/compactor_block.go (hunk 2 above)
Comment (mdisibio) [reply tempodb/encoding/compactor_block.go]: Good catch.  Was thinking the appendBuffer is flushed by the compactor block, but the appender is not.

**Review by joe-elliott — APPROVED** [review]

**Pull request merged**